frontend/__tests__/components/AddBookModal/add-book-modal.test.js

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import { render, screen, userEvent, waitFor } from "../../../test-utils";
+import Button from "../../../components/Button/Button";
+import Text from "../../../components/Text/Text";
+import Modal from "../../../components/Modal/Modal";
+import AddBookModal from "../../../components/AddBookModal/AddBookModal";
+
+jest.mock("../../../components/Modal/Modal");
+describe("Add Book Modal", () => {
+  let onCloseMock;
+  let onSubmitMock;
+  beforeEach(() => {
+    onCloseMock = jest.fn();
+    onSubmitMock = jest.fn();
+    Modal.mockImplementation(
+      ({
+        title,
+        children,
+        onClose,
+        onSubmit,
+        cancelText = "Cancel",
+        submitText = "Submit",
+      }) => (
+        <>
+          <Text>{title}</Text>
+          {children}
+          <Button onClick={onClose}>{cancelText}</Button>
+          <Button onClick={onSubmit}>{submitText}</Button>
+        </>
+      )
+    );
+    render(<AddBookModal onSubmit={onSubmitMock} onClose={onCloseMock} />);
+  });
+  it("should render add book modal component", () => {
+    expect(screen.getByText("Add New Book")).toBeInTheDocument();
+    expect(screen.getByText("Cancel")).toBeInTheDocument();
+    expect(screen.getByText("Add Book")).toBeInTheDocument();
+  });
```

Comment:
Once again, I think you really need to take a look at that front end testing best practices.